app/src/full/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -393,6 +411,29 @@ class MessagingService : FirebaseMessagingService() {
                 if (title == TURN_ON)
                     bluetoothAdapter.enable()
             }
+            COMMAND_NAVIGATE -> {
+                try {
+                    var mode = data["channel"]
+                    val validModes = listOf("d", "b", "l", "w")
+                    if (mode !in validModes)
+                        mode = "d"
+                    val navigate = "google.navigation:q=$title&mode=$mode"
+                    val mapsIntentUri = Uri.parse(navigate)
+                    val mapIntent = Intent(Intent.ACTION_VIEW, mapsIntentUri)
+                    mapIntent.setPackage("com.google.android.apps.maps")
+                    mapIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                    startActivity(mapIntent)
```

Comment:
I'm not sure we want to hard link to google navigation.  Maybe use the common intent?  I know it might not be identical but this way if someone doesn't use Google Maps this will still work for them.
https://developer.android.com/guide/components/intents-common#ViewMap

Reply:
Hmm ok so a special command that starts an activity with `Intent.ACTION_VIEW` preset?

Reply:
ok think i got it generic enough now lol